### PR TITLE
Improve logging for indexer

### LINKED ISSUE #559: Improve logging for indexer.
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Enrich the logs used during indexing to make tracing and debbuging easier.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by olsi-qose**
## Description
Enrich logs in order to make debug and tracing process easier.

## Related Issue
closes #559

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Model/Indexer/AbstractIndexer.php [hunk 1]
```diff
@@ -116,15 +116,76 @@ public function __construct(
...
+
+    /**
+     * @inheritdoc
+     * @throws Exception
+     */
+    public function executeRow($id)
+    {
+        $indexerId = $this->getIndexerId();
+        $message = sprintf(
+            'Begin a row  reindex for indexer "%s" for "%s"',
```
Comment (supercid) [inline Model/Indexer/AbstractIndexer.php]: Small one
```suggestion
            'Begin a row reindex for indexer "%s" for "%s"',
```

File: Model/Indexer/AbstractIndexer.php [hunk 2]
```diff
@@ -116,15 +116,76 @@ public function __construct(
...
+
+    /**
+     * @inheritdoc
+     * @throws Exception
+     */
+    public function executeList(array $ids)
+    {
+        $indexerId = $this->getIndexerId();
+        $idCount = count($ids);
+        $message = sprintf(
```
Comment (hannunosto) [inline Model/Indexer/AbstractIndexer.php]: Let's add here the total number of lines in cl table as well.

Comment (mridang) [reply Model/Indexer/AbstractIndexer.php]: It shouldn't be here. I'll explain.

File: Util/IndexerChangeLog.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Util;
```
Comment (mridang) [inline Util/IndexerChangeLog.php]: This should stay with the indexer code. This is not a general-purpose utility.

File: Model/Indexer/Provider/ChangeLogProvider.php [hunk 4]
```diff
@@ -34,27 +34,38 @@
...
  */
 
-namespace Nosto\Tagging\Util;
+namespace Nosto\Tagging\Model\Indexer\Provider;
 
-use Magento\Framework\App\ObjectManager;
 use Magento\Framework\App\ResourceConnection;
 
-class IndexerChangeLog
+class ChangeLogProvider
```
Comment (hannunosto) [inline Model/Indexer/Provider/ChangeLogProvider.php]: I would call this `ChangeLogService` or `IndexerStatusService`. We can later on add the cleaning of changelogs here as well.

File: Model/Indexer/Provider/ChangeLogProvider.php [hunk 5]
```diff
@@ -34,27 +34,38 @@
...
     {
-        $objectManager = ObjectManager::getInstance(); // Instance of object manager
-        /** @var ResourceConnection $resource */
-        $resource = $objectManager->get('Magento\Framework\App\ResourceConnection');
-        $connection = $resource->getConnection();
-        $sql = 'SELECT count(*) AS total FROM nosto_index_product_invalidate_cl';
-        $result = $connection->fetchAll($sql);
-        if (is_array($result) && count($result)>0){
-            return $result[0];
+        $connection = $this->resourceConnection->getConnection();
```
Comment (hannunosto) [inline Model/Indexer/Provider/ChangeLogProvider.php]: This looks a bit off with plain SQL queries. Check the Mview module: https://github.com/magento/magento2/tree/2.3/lib/internal/Magento/Framework/Mview. I think most of the stuff we need has been implemented already in that module.

**Comment by hannunosto** [comment]
We'll continue here: https://github.com/Nosto/nosto-magento2/pull/568

**Pull request closed without merging**